src/Ledger.Net/LedgerManager.cs

```diff
@@ -83,16 +83,32 @@ public LedgerManager(IHidDevice ledgerHidDevice, ICoinUtility coinUtility, Error
         #region Private Methods
         private async Task WriteRequestAsync<TWrite>(TWrite message) where TWrite : RequestBase
         {
-            var packetIndex = 0;
-            byte[] data = null;
-            using (var memoryStream = new MemoryStream(message.ToAPDU()))
+            var commandIndex = 0;
```

Comment:
Hey @edeharo . I don't think I understand this properly. I understand that chunks of data are broken up in to 256 byte chunks, and then further in to 64 byte chunks. However, if I read this correctly, it looks as though it will only send through one 255 byte chunk, and then on the second time, and every subsequent time it will try to read from the device. However, then the data is read, it is added to a response object, and nothing is done with this object. the response is never returned. It is only used to check for success. This doesn't seem quite right to me unless I'm not understanding correctly.

Reply:
@edeharo 

This is how Nicolas Dorier has broken up his code to achieve this:

https://github.com/LedgerHQ/ledger-dotnet-api/blob/2f4a7d0e0effc5d510c24558476bb58f68b88e7e/LedgerWallet/Transports/HIDTransportBase.cs#L87

Reply:
When I run the SignChunkedEthereumTransaction your branch, I get a status code of  27264.

Reply:
@edeharo this is my attempt at a fix https://github.com/MelbourneDeveloper/Ledger.Net/tree/fix-trx-signature-christian but it still doesn't work.

Reply:
Hi @MelbourneDeveloper 
Although I didn't analize the whole ethereum application https://github.com/LedgerHQ/ledger-app-eth , if you look at it, you will see that every time you send a signature request, the application tries to decode the signature request ( https://github.com/LedgerHQ/ledger-app-eth/blob/master/src_common/ethUstream.c , processTxInternal function), after performing some checks to ensure that the request honor the expected format and if the request isn't still complete (you didn't send the last chunk) the application returns the status 0x9000 indicating that the partial request was successfully processed. After the last chunk  was successfully processed, the application proceeds to calculate the signature and send it as response.

To me it doesn't make sense to return the partial responses (0x9000) since they only tell you if the data chunk was successfully processed, but of course, it is needed to check the SW in order to know if you can keep going sending the next chunk.

Again, although I did not analyze the eth app completely, if you take a look at the data that is sent when requesting a signature, you will see that it looks like this: SHA3(Trx Data) + Trx Data (https://github.com/Nethereum/Nethereum/blob/05e97108c1c07e692d032786ac04b4730fe7dba7/src/Nethereum.ABI/FunctionEncoding/FunctionCallEnconder.cs),  so that,  the way that the app knows that there are coming more data, seems to be by ensuring that Trx Data honor the transaction format https://github.com/ethereum/pyethereum/blob/develop/ethereum/transactions.py#L22 and then comparing the SHA3(Trx Data) already sent and the SHA3(Trx Data) calculated by itself. 

Regarding the status code 27264 (0x6A80) , did you activate the "Contract Data" option in your device ? If you didn't, that is the reason you get the error (https://github.com/LedgerHQ/ledger-app-eth/blob/master/src_genericwallet/main.c  , line 1816). 
From the Nethereum perspective, I think that this is something that could be improved by @juanfranblanco, there is a command that allows you to know if that option is active (you should add this command to your library), then, he knows when a contract transaction is about to be signed, meaning that he could raise an exception before send the command to the wallet.

Reply:
@edeharo excellent. I think you're right about all of that. I see that in your branch you were creating a result on each chunk to check the status code. That's not necessary though because the status code can be checked without doing that.

Anyway, the main point is that I needed to turn on Contract Data on the device. As soon as I did that, the unit test passed in my branch. 

I will implement checking each chunk's status code as you recommended and do some testing, then roll out the fix. 

Did you have a look at this branch? https://github.com/MelbourneDeveloper/Ledger.Net/compare/fix-trx-signature-christian . Can you see any issues with it?

Reply:
@edeharo , it's almost finished. Every test passes except GetAddressAnyBitcoinApp. The reason this test fails is that ToAPDUChunks doesn't yield any results. There is something wrong with that method. I will need to debug that some other time, but chunked transactions are almost there.

Reply:
@MelbourneDeveloper sorry about the bug, I just pushed the fix that solves it.

Reply:
@MelbourneDeveloper I didn't see any issues on your branch, it looks good!

Reply:
@edeharo thankyou! I cherry picked that commit and it worked perfectly. Now, the only question is how to handle getting this branch back in to master. I wanted to approve this pull request, but it seems Github has no option to approve without directly merging. So, I if you want to grab my branch, and then submit it back this repo as a new pull request, I will accept that pull request and merge it.

Otherwise I can just directly merge my branch in to master. Your commits (contribution) will still be there, but there will be no history of the pull request. Let me know so I can get this in to master ASAP.

Reply:
@MelbourneDeveloper Glad to hear that. No worries for my branch, and go ahead and merge your branch. Eager to update my project with your next release !